java/src/main/java/ai/onnxruntime/OrtSession.java

```diff
@@ -441,7 +484,10 @@ private native OnnxModelMetadata constructMetadata(
    * <p>The SessionOptions object must not be closed until all sessions which use it are closed, as
    * otherwise it could release resources that are in use.
    */
-  public static class SessionOptions implements AutoCloseable {
+  public static class SessionOptions extends NativeObject {
+
+    static final OrtSession.SessionOptions DEFAULT_SESSION_OPTIONS =
```

Comment:
Is this a good idea? I think the session options isn't designed to be shared between different sessions. I realise the default one doesn't seem to currently hold any shared resources, but that's not to say it won't in the future, and such a change wouldn't be visible from the Java API.

Reply:
Yeah I'm fine creating a new session options each time, since they are very minimal.